### PR TITLE
File comparison test does not check __CLASS_HASH

### LINKED ISSUE #2548: We should ignore `__CLASS_HASH` changes on Schema tests
On `SchemaTest`, we check if the generated classes and the old classes are the same, to detect api breaks. We compare the files and require all the lines to be the same. As the `__CLASS_HASH = <hash>` line is not part of the api, it should be ignored when comparing the two files.

### PR DESCRIPTION
**Pull request by OStevan**
**Goals (and why)**:
Fixes #2548, when comparing two files in SchemaTest ignore __CLASS_HASH. Comparing two generated class source files to detect API breaks checks also the __CLASS_HASH which does not contribute to the API.

**Implementation Description (bullets)**:
- check if the files have the same size
- stream the files contents per line to detect differences ignoring the lines with __CLASS_HASH

**Testing (What was existing testing like?  What have you done to improve it?)**:
This PR improves tests.

**Concerns (what feedback would you like?)**:
Source code readability

**Where should we start reviewing?**:
n/a(single file changed)

**Priority (whenever / two weeks / yesterday)**:
whenever

### CODE REVIEW

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java [hunk 1]
```diff
@@ -209,10 +214,31 @@ private void checkIfFilesAreTheSame(List<String> generatedTestTables) {
...
             File expectedFile = new File(EXPECTED_FILES_FOLDER_PATH, generatedFilePath);
             File actualFile = new File(testFolder.getRoot(), generatedFilePath);
-            assertThat(actualFile).hasSameContentAs(expectedFile);
+
+            assertThat(expectedFile.length()).isEqualTo(actualFile.length());
+
+            try (Stream<String> expectedFileStream = java.nio.file.Files
+                        .lines(Paths.get(EXPECTED_FILES_FOLDER_PATH, generatedFilePath));
+                Stream<String> actualFileStream = java.nio.file.Files
+                        .lines(Paths.get(testFolder.getRoot().getPath(), generatedFilePath));
```
Comment (jeremyk-91) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: There's a bit of duplication here with getting the paths here and in getting the Files above. I think you can just call `File.toPath()` on the files created from lines 217/218

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java [hunk 2]
```diff
@@ -209,10 +214,31 @@ private void checkIfFilesAreTheSame(List<String> generatedTestTables) {
...
+                Stream<Boolean> zipped = Streams.zip(
+                        expectedFileStream,
+                        actualFileStream,
+                        (first, second) -> {
+                            if (first.equals(second))
+                                return true;
+                            return first.contains("__CLASS_HASH") && second.contains("__CLASS_HASH");
```
Comment (jeremyk-91) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: nit: The `__CLASS_HASH` string should probably be a constant. Also I normally would just write this check one line, though that's more of a matter of style preference

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java [hunk 3]
```diff
@@ -209,10 +214,31 @@ private void checkIfFilesAreTheSame(List<String> generatedTestTables) {
...
             File actualFile = new File(testFolder.getRoot(), generatedFilePath);
-            assertThat(actualFile).hasSameContentAs(expectedFile);
+
+            assertThat(expectedFile.length()).isEqualTo(actualFile.length());
+
+            try (Stream<String> expectedFileStream = java.nio.file.Files
+                        .lines(Paths.get(EXPECTED_FILES_FOLDER_PATH, generatedFilePath));
+                Stream<String> actualFileStream = java.nio.file.Files
+                        .lines(Paths.get(testFolder.getRoot().getPath(), generatedFilePath));
+                Stream<Boolean> zipped = Streams.zip(
```
Comment (jeremyk-91) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: nit: probably best to say what the booleans mean (maybe something like `correspondingLinesMatchOrAreHashes`)?

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java [hunk 4]
```diff
@@ -209,10 +214,31 @@ private void checkIfFilesAreTheSame(List<String> generatedTestTables) {
...
+                                return true;
+                            return first.contains("__CLASS_HASH") && second.contains("__CLASS_HASH");
+                        })) {
+
+                assertThat(zipped
+                        .anyMatch(elem -> !elem))
+                        .isFalse();
+            } catch (IOException e) {
+                Assertions.fail("Exception on stream creation", e);
+            }
```
Comment (jeremyk-91) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: Usually we just add the exception to the method signature (junit will treat a test as a failure if an exception gets thrown out of the test method)

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java [hunk 5]
```diff
@@ -209,10 +214,31 @@ private void checkIfFilesAreTheSame(List<String> generatedTestTables) {
...
+                        actualFileStream,
+                        (first, second) -> {
+                            if (first.equals(second))
+                                return true;
+                            return first.contains("__CLASS_HASH") && second.contains("__CLASS_HASH");
+                        })) {
+
+                assertThat(zipped
+                        .anyMatch(elem -> !elem))
+                        .isFalse();
```
Comment (jeremyk-91) [inline atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: AssertJ does have pretty nice support for streams, so you can do something like `assertThat(zipped).allMatch(elem -> elem);` directly

Also note that we're asserting `not exists not elem`, which is equivalent to `for-all elem` which I think is clearer

File: atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java (hunk 4 above)
Comment (OStevan) [reply atlasdb-client/src/test/java/com/palantir/atlasdb/table/description/SchemaTest.java]: Exception is raised inside a lambda so removing a try is not an option.

**Pull request merged**